client/version/self_update.go

```diff
@@ -0,0 +1,54 @@
+package version
+
+import (
+	"github.com/kowala-tech/kcoin/client/node"
+	"github.com/kowala-tech/kcoin/client/internal/debug"
+	"github.com/kowala-tech/kcoin/client/log"
+	"time"
+)
+
+type SelfUpdater struct {
+	repository string
+	stack      *node.Node
+	logger     log.Logger
+}
+
+func NewSelfUpdater(repository string, stack *node.Node, logger log.Logger) *SelfUpdater {
+	return &SelfUpdater{
+		repository: repository,
+		stack:      stack,
+		logger:     logger,
+	}
+}
+
+func (su *SelfUpdater) Run() {
+	updater, err := NewUpdater(su.repository, su.logger)
+	if err != nil {
+		su.logger.Warn("error starting update for selfupdate, selfupdate disabled")
+		return
+	}
+
+	for range time.Tick(time.Minute / 6) {
```

Comment:
I see were that might happen, here I think at each iteration uses same ticker, so only one ticker is created on app start. Never the less going to prove or otherwise :)

Reply:
I think so, only one timer+channel should leak, but it shouldn't be a problem.